### PR TITLE
`Development`: Remove no longer required secret push token for Jenkins setups

### PR DESCRIPTION
**Pull request by b-fein**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a [very good performance](https://docs.artemis.cit.tum.de/dev/guidelines/performance/) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).

#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server configured with **Gitlab** and **Jenkins**.

### Motivation and Context

Since #6682 the builds are no longer triggered in Jenkins from GitLab directly. Instead, Artemis triggers the build by using the regular API authentication of Jenkins.
The secret token was only used to allow GitLab to trigger builds directly in Jenkins. Since this feature is no longer used it can be removed.
It also does not make sense to keep it since it is not required for LocalVC+Jenkins setups.

### Description

Removed references to the no longer needed `secret-push-token` config option.

As should be clear from the server code changes, the secret token is only used when creating the initial build plan. When triggering builds in the Jenkins, Artemis does not use it.

### Steps for Testing
Prerequisites:
- a GitLab+Jenkins or LocalVC+Jenkins setup

1. Create a new programming exercise.
2. When making changes to the template or solution repositories the Jenkins build should run and updated results show up in Artemis.

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage
unchanged

### Screenshots
n/a

@coderabbitai ignore

### CODE REVIEW

**Comment by dfuchss** [comment]
I'll test the PR asap

**Review by dfuchss — DISMISSED** [review]
Changes make sense.

**Review by JohannesWt — APPROVED** [review]
Code 👍

File: docs/dev/setup/jenkins-gitlab.rst [hunk 1]
```diff
@@ -691,6 +691,18 @@ Start Jenkins
                user: your.chosen.username
                password: your.chosen.password
 
+11. In a local setup, you have to disable CSRF otherwise some API endpoints will return HTTP Status 403 Forbidden.
```
Comment (b-fein) [inline docs/dev/setup/jenkins-gitlab.rst:694]: Note: this is not new, but got lost when removing the secret push token section in #9266.

**Review by dfuchss — APPROVED** [review]

**Review by magaupp — APPROVED** [review]
Code LGTM

**Review by SimonEntholzer — APPROVED** [review]
code changes make sense :+1:

**Comment by b-fein** [comment]
I confirmed on a test system that the web-hook from GitLab to Jenkins does indeed no longer exist and did not before this PR either. Therefore, the Jenkins secret token is never used to trigger build plans in existing systems already. This ensures currently running GitLab+Jenkins setups will not break.

Since I don’t think anyone else testing PRs is using the deprecated GitLab+Jenkins setup any more (and it is definitely not worth setting it up just for this), I will count this as the manual test and mark this PR as ready to merge.

**Review by krusche — APPROVED** [review]

**Pull request merged**